packages/components/src/form-token-field/suggestions-list.js

```diff
@@ -1,158 +1,150 @@
 /**
  * External dependencies
  */
-import { map } from 'lodash';
-import scrollIntoView from 'dom-scroll-into-view';
+import { map, noop } from 'lodash';
+import scrollView from 'dom-scroll-into-view';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { withSafeTimeout } from '@wordpress/compose';
 
-class SuggestionsList extends Component {
-	constructor() {
-		super( ...arguments );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.bindList = this.bindList.bind( this );
-	}
+const emptyList = Object.freeze( [] );
 
-	componentDidUpdate() {
+const handleMouseDown = ( e ) => {
+	// By preventing default here, we will not lose focus of <input> when clicking a suggestion
+	e.preventDefault();
+};
+
+function SuggestionsList( {
+	selectedIndex,
+	scrollIntoView,
+	match = '',
+	onHover = noop,
+	onSelect = noop,
+	suggestions = emptyList,
+	displayTransform,
+	instanceId,
+	setTimeout,
+} ) {
+	const [ scrollingIntoView, setScrollingIntoView ] = useState( true );
```

Comment:
Oops! you're right 💯 

Correct me If I'm wrong here, I checked variable here https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/form-token-field/index.js#L50
and based on that I'm agreeing its value should be false.

Reply:
My reasoning was that, prior to your refactor, `this.scrollingIntoView` was not initialised to anything — and therefore it held a value of `undefined`. Since we're refactoring the code to `useState` and we want that variable to always hold a boolean value, then we should use `false` (since `undefined` is a falsy value)

Hope this is clear enough!